avocado/core/test.py

```diff
@@ -133,6 +133,98 @@ def str_filesystem(self):
+
+    This function is currently a thin wrapper (almost an alias) for
+    :attr:`avocado.core.test.Test.datadir`.  The idea is just to give
+    consistent naming across the various data dir, and prepare for the
+    removal of :attr:`avocado.core.test.Test.datadir`.
+
+    :param test: the test to locate its file-level data directory
+    """
+    # TODO: on removal of Test.datadir, move its code over here
+    return test.datadir
+
+
+def get_test_data_dir(test):
+    """
+    Returns the path to the test-level data directory
+
+    For an INSTRUMENTED test, implemented by a method named `test`, on
+    a class named `SleepTest`, hosted at a file located at
+    /usr/share/avocado/tests/sleeptest.py, the test-level data directory
+    is /usr/share/avocado/tests/sleeptest.py.data/SleepTest.test
+
+    For a SIMPLE test, the test is the file, so this is the same as
+    :func:`get_file_data_dir`.
+
+    :param test: the test to locate its file-level data directory
+    """
+    file_data_dir = get_file_data_dir(test)
+    if file_data_dir is None:
+        return None
+
+    if isinstance(test, SimpleTest):
+        return file_data_dir
+
+    test_level_data_dir = "%s.%s" % (test.__class__.__name__,
+                                     test._testMethodName)
+    test_level_data_dir = astring.string_to_safe_path(test_level_data_dir)
+    return os.path.join(file_data_dir, test_level_data_dir)
+
+
+def get_variant_data_dir(test):
+    """
+    Returns the path to the variant-level data directory
+
+    For an INSTRUMENTED test, implemented by a method named `test`, on
+    a class named `SleepTest`, hosted at a file located at
+    /usr/share/avocado/tests/sleeptest.py, and with variant name
+    "foo-ffff", the variant-level data directory is
+    /usr/share/avocado/tests/sleeptest.py.data/SleepTest.test;foo-ffff
+
+    :param test: the test to locate its file-level data directory
+    """
+    test_data_dir = get_test_data_dir(test)
+    if test_data_dir is None or test.name.variant is None:
+        return None
+
+    variant_dirname = astring.string_to_safe_path(test.name.variant)
+    return os.path.join(test_data_dir, variant_dirname)
+
+
+def get_simplified_variant_data_dir(test):
+    """
+    Returns the path to the simplified variant-level data directory
+
+    This is a directory that omits the 4 digits hashes that acompany
+    variant names.  The reason for this is that usually, test variant
+    data tends to change at a much slower pace than the variant hashes,
+    which also considers keys and values content.
+
+    This implementation simply strips the hash and separator, that is,
+    for variant-level data dir "t.py:Klass.test;foo-ffff", it returns
+    "t.py:Klass.test;foo"
+
+    :param test: the test to locate its file-level data directory
+    """
+    variant_data_dir = get_variant_data_dir(test)
+    if variant_data_dir is None:
+        return None
+
+    return variant_data_dir[:-5]
```

Comment:
This only works for varianter plugins that use `avocado.core.varianter.variant_to_str`, but variant id can be any arbitrary string (eg. 1, 2, 3). Besides the whole purpose of the hash of the content is to change the datadir in case any value is changed. I know sometimes it might not be intended, but it's the safest approach. Anyway if you want to keep the possibility, then we need a special way of getting the variant-name, but I'd start without such support and if users ask for it we'd have specific use case, therefor probably better results.

Reply:
OK, let's drop that for now.  I think it'll make a quick comeback, though.

Reply:
Yep, could be, but it should be implemented on the varianter level, not just hacked in here...

Reply:
PS: Feel free to even add trello card to do that... I think it'll be useful.